dynawo/sources/Simulation/DYNSimulation.cpp

```diff
@@ -483,25 +484,29 @@ Simulation::loadDynamicData() {
   dyd_->setRootDirectory(context_->getInputDirectory());
   dyd_->setParametersReference(referenceParameters_);
 
-  if (iidmFile_ != "") {
-    data_ = DataInterfaceFactory::build(DataInterfaceFactory::DATAINTERFACE_IIDM1, iidmFile_);
-    if (criteriaCollection_)
-      data_->configureCriteria(criteriaCollection_);
+  if (!data_) {
+    if (iidmFile_ != "") {
+      data_ = DataInterfaceFactory::build(DataInterfaceFactory::DATAINTERFACE_IIDM1, iidmFile_);
+    } else {
+      dyd_->initFromDydFiles(dydFiles_);
+      if (activateCriteria_) {
+        Trace::warn() << DYNLog(CriteriaDefinedButNoIIDM) << Trace::endline;
+      }
+      return;
+    }
+  }
+  if (criteriaCollection_)
+    data_->configureCriteria(criteriaCollection_);
```

Comment:
I think this one should be done only once, we might have an issue with systematic analysis otherwise

Reply:
Where else is it done in the code ? I didn't change the logic in dynawo and it is not done in DFL

Reply:
it will be used later in systematic analysis, we can postpone this to this usage but my guess is that it will be an issue

Reply:
I don't get. dynawo-interface performs systematic analysis, correct ? And it is using dynawo simulation the same way DFL is doing it.

Reply:
yes, but later we will inject directly the data interface in both dyn-itf and DFL